net/ieee8021x/files/ieee8021xclient.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+
+[ -n "$INCLUDE_ONLY" ] || {
+	. /lib/functions.sh
+	. /lib/functions/network.sh
+	. ../netifd-proto.sh
+	init_proto "$@"
+}
+
+remove_empty_line() {
+	echo "$1" | sed '/^[[:space:]]*$/d'
+}
+
+ieee8021xclient_exitcode_tostring() {
+	local errorcode=$1
+	[ -n "$errorcode" ] || errorcode=5
+
+	case "$errorcode" in
+		0) echo "OK" ;;
+		1) echo "FATAL_ERROR" ;;
+		5) echo "USER_REQUEST" ;;
+		*) echo "UNKNOWN_ERROR" ;;
+	esac
+}
+
+_wpa_supplicant_common() {
+	local ifname="$1"
+
+	_config="/var/run/wpa_supplicant-$ifname.conf"
+}
+
+proto_ieee8021xclient_setup() {
+	local cfg="$1"
+	local ifname="$2"
+
+	local _config
+	_wpa_supplicant_common "$ifname"
+
+	local eapol_version
+	local identity anonymous_identity password
+	local ca_cert client_cert private_key private_key_passwd dh_file subject_match
+	local phase1 phase2 ca_cert2 client_cert2 private_key2 private_key_passwd2 dh_file2 subject_match2
+	local eap_workaround
+	json_get_vars eapol_version
+	json_get_vars identity anonymous_identity password
+	json_get_vars ca_cert client_cert private_key private_key_passwd dh_file subject_match
+	json_get_vars phase1 phase2 ca_cert2 client_cert2 private_key2 private_key_passwd2 dh_file2 subject_match2
+	json_get_vars eap_workaround
+
+	# launch
+	cat > "${_config}" << EOF
```

Comment:
What do we do if the interface is not there. And therefore no configuration can be written. 

We should return an error if [this file](https://github.com/openwrt/packages/pull/15983/files#diff-96f9a3e11e1c3382cb7bf3844ee1d26e86577c2a080bae547c403a1bf4313b9eR29) is not there

Reply:
What do you mean by 'no configuration can be written'?

The _config file is not supposed to be here before we create it.

Reply:
ppp proto does not check for present of interface too? https://github.com/openwrt/openwrt/blob/master/package/network/services/ppp/files/ppp.sh#L221

Reply:
Just because it's not done there doesn't mean it's right.
But it was just a suggestion.
I can live without the check.
Just thought that it might simplify the debugging task.

Reply:
I can't image why proto_X_setup will not be given a valid interface.

Nor I can find an example on how to do the check...